packages/react-instantsearch/src/core/InstantSearch.js

```diff
@@ -137,12 +138,22 @@ class InstantSearch extends Component {
   }
 
   getChildContext() {
+    // If not already cached, cache the bound methods so that we can forward them as part
+    // of the context.
+    if (!this._aisContextCache) {
+      this._aisContextCache = {
+        ais: {
+          onInternalStateUpdate: this.onWidgetsInternalStateUpdate.bind(this),
+          createHrefForState: this.createHrefForState.bind(this),
```

Comment:
Yes but bind is only mandatory because we export those. So this way makes the bind more explicit and localised. I didn't choose to do it the way you propose because I thought that it wasn't needed.

We avoid the assignment construct in class because jsdoc is as of now unable to parse it :(